Add Origin-Agent-Cluster article (#4448)
GoogleChrome/web.dev: src/site/content/en/blog/origin-agent-cluster/index.md

---
layout: post
title: Requesting performance isolation with the Origin-Agent-Cluster header
subhead: A new HTTP response header to limit domain-wide scripting and request dedicated resources from the browser.
authors:
  - domenic
description: >
  The Origin-Agent-Cluster header cuts off synchronous access to other origins on the same domain, and hints to the browser to give your origin dedicated resources.

> date: 2021-01-11

Reviewer: ```suggestion
date: 2021-01-30
```

> hero: hero.jpg

Reviewer: Please optimize this with tinyjpg.com (to keep the size of the repository down)

P.S. I like the "cluster" reference

Author: Thanks, and done! I feel like the guidance on https://web.dev/handbook/markup-media/ used to say no optimization, but now it's updated...

alt: A cluster of grapes
tags:
  - blog # blog is a required tag for the article to show up in the blog.
  - performance
  - memory
---

> 

Reviewer: Please break all lines at 100 characters. If you're using VS Code, [Rewrap](https://marketplace.visualstudio.com/items?itemName=stkb.rewrap) makes it pretty easy to do. The [Docs To X](https://gsuite.google.com/u/0/marketplace/app/docs_to_x/243920501110?hl=en&pann=docs_addon_widget) Google Docs Add-On also supports this.

Author: Done!

* The browser might decide it's not worth the overhead of a separate process. For example, on low-memory Android devices, or in Android WebView, Chrome uses as few processes as possible.

* The browser might want to respect the request that the `Origin-Agent-Cluster` header indicates, but it could do so using different isolation technology than processes. For example, Chrome is [exploring](https://docs.google.com/document/d/12wEWJsZmxVnNwVGuxuEJF4922OWUr4fCs1xKHi9mTiI/edit?usp=sharing) using threads instead of processes for this sort of performance isolation.

* The user, or code running on a different site, might have already navigated to a site-keyed page on your origin, causing the [consistency guarantee](#how-to-use) to kick in and the `Origin-Agent-Cluster` header to be ignored entirely.

For these reasons, it's important not to think of origin-keyed agent clusters as a security feature. Instead, it's a way of helping the browser prioritize resource allocation, by hinting that your origin would benefit from dedicated resources (and that you're willing to give up [certain features](#limitations) in exchange).

## Feedback

> We'd love to hear from you if you're using, or considering using, the `Origin-Agent-Cluster` header. Your public interest and support helps the Chrome team prioritize features and show other browser vendors how important they are. Tweet at [@ChromiumDev](https://twitter.com/chromiumdev) and let us know your thoughts and experiences.

Reviewer: ```suggestion
Chrome love to hear from you if you're using, or considering using, the `Origin-Agent-Cluster` header. Your public interest and support helps the Chrome team prioritize features and show other browser vendors how important they are. Tweet at [@ChromiumDev](https://twitter.com/chromiumdev) and let Chrome DevRel know your thoughts and experiences.
```
Avoiding "we" and just making it explicit instead

If you have more questions about the specification, or the details of how the feature works, you can file an issue on the [HTML Standard GitHub repository](https://github.com/whatwg/html). And if you encounter any issues with Chrome's implementation, you can file a bug at [new.crbug.com](https://bugs.chromium.org/p/chromium/issues/entry?components=Internals%3ESandbox%3ESiteIsolation) with the Components field set to `Internals>Sandbox>SiteIsolation`.

## Learn more

> To learn more about origi-keyed agent clusters, you can dive into the details at these links:

Reviewer: ```suggestion
To learn more about origin-keyed agent clusters, you can dive into the details at these links:
```

---
layout: post
title: Requesting performance isolation with the Origin-Agent-Cluster header
subhead: >
  A new HTTP response header to limit domain-wide scripting and request dedicated resources from the
  browser.
authors:
  - domenic
description: >
  The Origin-Agent-Cluster header cuts off synchronous access to other origins on the same domain,
  and hints to the browser to give your origin dedicated resources.

> date: 2021-01-30

Reviewer: ```suggestion
date: 2021-02-01
```
